### ISSUE TITLE
NoGaps command not producing 24 hour image

### ISSUE THREAD

**Issue by joderetic** [QutEcoacoustics/audio-analysis#303]
## Describe the bug
Indices map2.png not showing 24 hour period for processed data when using NoGaps command. In this case it is showing a 30min period only.

## Expected behavior 
The indices map2.png image should show a 24 hour period for the processed data using the NoGaps command. 

### Instructions

- [x] I have followed the [instructions for reporting a bug](https://github.com/QutEcoacoustics/audio-analysis/blob/master/docs/bug_report.md)

### I have attached a log:

<details><summary>Inline log</summary>
2020-03-19T11:37:58.2976118+11:00 [1] INFO  CleanLogger - QUT Ecoacoustics Analysis Programs - version 20.2.0.99 (RELEASE build, 2020-02-11T04:22:19.0353321Z)
Git branch-version: master-8bc900890b804d5262fcecae2f05c34a9c5bb533, DirtyBuild:False, CI:000
Copyright 2020 QUT
2020-03-19T11:37:58.3132337+11:00 [1] INFO  LogFileOnly - Executable called with these arguments: 
"D:\AP\AnalysisPrograms.exe" audio2csv D:\Recordings\JoDeretic_DoctorsSwamp\S1-SP_20170925_013000.wav D:\Scripts\..\Configs\Towsey.Acoustic.30.yml Output\DoctorsIndices\S1-SP_20170925_013000.wav --quiet

2020-03-19T11:37:58.4713466+11:00 [1] INFO  CleanLogger - # PROCESS LONG RECORDING
2020-03-19T11:37:58.4713466+11:00 [1] INFO  CleanLogger - # DATE AND TIME: 19/03/2020 11:37:58 AM
2020-03-19T11:37:58.4713466+11:00 [1] WARN  AnalyseLongRecording - No temporary directory provided, using output directory
2020-03-19T11:37:58.4713466+11:00 [1] INFO  CleanLogger - # Recording file:      D:\Recordings\JoDeretic_DoctorsSwamp\S1-SP_20170925_013000.wav
2020-03-19T11:37:58.4713466+11:00 [1] INFO  CleanLogger - # Configuration file:  D:\Scripts\..\Configs\Towsey.Acoustic.30.yml
2020-03-19T11:37:58.4713466+11:00 [1] INFO  CleanLogger - # Output folder:       Output\DoctorsIndices\S1-SP_20170925_013000.wav
2020-03-19T11:37:58.4713466+11:00 [1] INFO  CleanLogger - # Temp File Directory: Output\DoctorsIndices\S1-SP_20170925_013000.wav
2020-03-19T11:37:58.5714054+11:00 [1] INFO  AnalyseLongRecording - Using analyzer Towsey.Acoustic
2020-03-19T11:37:58.8560881+11:00 [1] INFO  LogFileOnly - Config file `D:\Configs\Towsey.Acoustic.30.yml` loaded:
{"LdSpectrogramConfig":{"FreqScale":"Linear","ColorMap1":"ACI-ENT-EVN","ColorMap2":"BGN-PMN-SPT","ColourFilter":-0.25,"BlueEnhanceParameter":0.0,"XAxisTicIntervalSeconds":3600.0,"YAxisTicInterval":1000,"ConfigPath":null,"GenericConfig":null},"TileOutput":false,"IndexCalculationDuration":30.0,"BgNoiseNeighborhood":5.0,"FrameLength":512,"LowFreqBound":1000,"MidFreqBound":8000,"FrequencyScale":0,"MinBandWidth":0.0,"MaxBandWidth":1.0,"MelScale":0,"IndexPropertiesConfig":"./IndexPropertiesConfig.yml","IndexProperties":null,"AnalysisName":null,"EventThreshold":0.2,"SegmentDuration":30.0,"SegmentOverlap":0.0,"ResampleRate":22050,"SaveIntermediateWavFiles":0,"SaveIntermediateCsvFiles":false,"SaveSonogramImages":0,"RequireDateInFilename":false,"ConfigPath":"D:\\Configs\\Towsey.Acoustic.30.yml","GenericConfig":{"SegmentDuration":"30","SegmentOverlap":"0","IndexCalculationDuration":"30.0","BgNoiseNeighborhood":"5","ResampleRate":"22050","FrameLength":"512","LowFreqBound":"1000","MidFreqBound":"8000","HighFreqBound":"11000","FrequencyScale":"Linear","SaveIntermediateWavFiles":"Never","SaveIntermediateCsvFiles":"false","SaveSonogramImages":"Never","SaveSonogramData":"false","TileOutput":"false","RequireDateInFilename":"false","IndexPropertiesConfig":"./IndexPropertiesConfig.yml","EventThreshold":"0.2","LdSpectrogramConfig":{"ColorMap1":"ACI-ENT-EVN","ColorMap2":"BGN-PMN-SPT","ColourFilter":"-0.25","XAxisTicIntervalSeconds":"3600","FreqScale":"Linear","YAxisTicInterval":"1000"}}}
2020-03-19T11:37:58.8873291+11:00 [1] INFO  LogFileOnly - Config file `D:\Configs\IndexPropertiesConfig.yml` loaded:
{"RankOrder":{"DataType":"int","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":240.0,"CalculateNormBounds":false},"StartOffset":{"DataType":"double","DefaultValue":0.0,"DoDisplay":false,"NormMin":0.0,"NormMax":200.0,"CalculateNormBounds":false},"StartOffsetMinute":{"DataType":"int","DefaultValue":0.0,"DoDisplay":false,"NormMin":0.0,"NormMax":200.0,"CalculateNormBounds":false},"ResultStartSeconds":{"DataType":"int","DefaultValue":0.0,"DoDisplay":false,"NormMin":0.0,"NormMax":200.0,"CalculateNormBounds":false},"SegmentDurationSeconds":{"DataType":"TimeSpan","DefaultValue":0.0,"DoDisplay":false,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"HighAmplitudeIndex":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":5.0,"CalculateNormBounds":false},"ZeroSignal":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"ClippingIndex":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":5.0,"CalculateNormBounds":false},"AvgSignalAmplitude":{"DataType":"double","DefaultValue":-80.0,"DoDisplay":true,"NormMin":-50.0,"NormMax":-10.0,"CalculateNormBounds":false},"BackgroundNoise":{"DataType":"double","DefaultValue":-80.0,"DoDisplay":true,"NormMin":-50.0,"NormMax":-10.0,"CalculateNormBounds":false},"Snr":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":50.0,"CalculateNormBounds":false},"AvgSnrOfActiveFrames":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":30.0,"CalculateNormBounds":false},"Activity":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"EventsPerSecond":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":5.0,"CalculateNormBounds":false},"AvgEventDuration":{"DataType":"TimeSpan","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1000.0,"CalculateNormBounds":false},"HighFreqCover":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.5,"CalculateNormBounds":false},"MidFreqCover":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.5,"CalculateNormBounds":false},"LowFreqCover":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.5,"CalculateNormBounds":false},"AcousticComplexity":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.4,"NormMax":0.7,"CalculateNormBounds":false},"TemporalEntropy":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.5,"CalculateNormBounds":false},"SqrtTempEntropy":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.7,"CalculateNormBounds":false},"LogTempEntropy":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.176,"CalculateNormBounds":false},"EntropyOfAverageSpectrum":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"EntropyOfVarianceSpectrum":{"DataType":"double","DefaultValue":0.0,"DoDisplay":false,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"EntropyOfPeaksSpectrum":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"EntropyOfCoVSpectrum":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"ClusterCount":{"DataType":"int","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":50.0,"CalculateNormBounds":false},"ThreeGramCount":{"DataType":"int","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1500.0,"CalculateNormBounds":false},"SptDensity":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":10.0,"CalculateNormBounds":false},"NDSI":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-1.0,"NormMax":1.0,"CalculateNormBounds":false},"NDSI-LM":{"DataType":"double","DefaultValue":0.0,"DoDisplay":false,"NormMin":-1.0,"NormMax":1.0,"CalculateNormBounds":false},"NDSI-MH":{"DataType":"double","DefaultValue":0.0,"DoDisplay":false,"NormMin":-1.0,"NormMax":1.0,"CalculateNormBounds":false},"EventsTotal":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":15.0,"CalculateNormBounds":false},"EventsTotalThresholded":{"DataType":"double","DefaultValue":0.0,"DoDisplay":false,"NormMin":5.0,"NormMax":25.0,"CalculateNormBounds":false},"ZeroCrossingFractionLeft":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"ZeroCrossingFractionRight":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"ChannelSimilarity":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"ChannelDiffDecibels":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-8.0,"NormMax":8.0,"CalculateNormBounds":false},"AverageDecibelBias":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-8.0,"NormMax":8.0,"CalculateNormBounds":false},"MedianDecibelBias":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-0.2,"NormMax":0.2,"CalculateNormBounds":false},"LowFreqDecibelBias":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-10.0,"NormMax":10.0,"CalculateNormBounds":false},"MidFreqDecibelBias":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-10.0,"NormMax":10.0,"CalculateNormBounds":false},"HighFreqDecibelBias":{"DataType":"double","DefaultValue":0.0,"DoDisplay":true,"NormMin":-5.0,"NormMax":5.0,"CalculateNormBounds":false},"FileName":{"DataType":"int","DefaultValue":0.0,"DoDisplay":false,"NormMin":0.0,"NormMax":1.0,"CalculateNormBounds":false},"ACI":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.4,"NormMax":0.7,"CalculateNormBounds":false},"BGN":{"DataType":"double[]","DefaultValue":-100.0,"DoDisplay":true,"NormMin":-100.0,"NormMax":-30.0,"CalculateNormBounds":false},"CVR":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.4,"CalculateNormBounds":false},"DIF":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":10.0,"CalculateNormBounds":false},"ENT":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.6,"CalculateNormBounds":false},"EVN":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":4.0,"CalculateNormBounds":false},"OSC":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.1,"NormMax":5.0,"CalculateNormBounds":true},"SUM":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":10.0,"CalculateNormBounds":false},"PMN":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":15.0,"CalculateNormBounds":false},"RHZ":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.1,"NormMax":1.0,"CalculateNormBounds":false},"RVT":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.1,"NormMax":0.4,"CalculateNormBounds":true},"RPS":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.4,"CalculateNormBounds":true},"RNG":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.0,"NormMax":0.4,"CalculateNormBounds":true},"SPT":{"DataType":"double[]","DefaultValue":0.0,"DoDisplay":true,"NormMin":0.05,"NormMax":0.5,"CalculateNormBounds":false}}
2020-03-19T11:37:58.9185723+11:00 [1] INFO  CleanLogger - # IndexProperties Cfg: D:\Configs\IndexPropertiesConfig.yml
2020-03-19T11:37:58.9185723+11:00 [1] INFO  AnalyseLongRecording - Minimum event threshold has been set to 0.2
2020-03-19T11:37:58.9185723+11:00 [1] WARN  Acoustics.Shared.FileDateHelpers - File date `20170925-013000` is ambiguous. The date is understood but no timezone offset could be found and a timezone offset hint was not provided.
2020-03-19T11:37:59.0866256+11:00 [1] INFO  AnalyseLongRecording - SegmentDuration=00:00:30, SegmentOverlap=00:00:00, ResampleRate=22050
2020-03-19T11:37:59.0866256+11:00 [1] INFO  CleanLogger - START ANALYSIS ...
2020-03-19T11:37:59.0866256+11:00 [1] INFO  AnalysisCoordinator - Analysis Coordinator will analyze a total of 30 seconds of audio
2020-03-19T11:37:59.1047307+11:00 [1] INFO  AnalysisCoordinator - Executing BeforeAnalyze
2020-03-19T11:37:59.1047307+11:00 [1] INFO  AnalysisPrograms.AcousticIndices - Image tiling is NOT enabled
2020-03-19T11:37:59.1203534+11:00 [1] INFO  AnalysisCoordinator - Analysis started in sequence.
2020-03-19T11:38:02.1937829+11:00 [1] INFO  AnalysisCoordinator - Completed segment 1/1
2020-03-19T11:38:02.1937829+11:00 [1] INFO  AnalysisCoordinator - Analysis complete, took 00:00:03.0737400.
2020-03-19T11:38:02.1937829+11:00 [1] INFO  CleanLogger - 
2020-03-19T11:38:02.1937829+11:00 [1] INFO  CleanLogger - START PROCESSING RESULTS ...
2020-03-19T11:38:02.3277549+11:00 [1] WARN  AnalysisPrograms.AcousticIndices - False color spectrograms were not rendered
2020-03-19T11:38:02.6128820+11:00 [1] WARN  AudioAnalysisTools.Indices.IndexDisplay - Indices not found: Ndsi, ResultMinute,
2020-03-19T11:38:02.6128820+11:00 [1] WARN  AudioAnalysisTools.Indices.IndexDisplay - Indices not plotted: EntropyOfVarianceSpectrum, FileName, ResultStartSeconds, SegmentDurationSeconds,
2020-03-19T11:38:02.6285087+11:00 [1] INFO  CleanLogger - INDICES CSV file(s) = S1-SP_20170925_013000__Towsey.Acoustic.Indices.csv
2020-03-19T11:38:02.6285087+11:00 [1] INFO  CleanLogger - 	Number of rows (i.e. minutes) in CSV file of indices = 1
2020-03-19T11:38:02.6285087+11:00 [1] INFO  CleanLogger - 
2020-03-19T11:38:02.6285087+11:00 [1] INFO  CleanLogger - An Events CSV file was NOT returned.
2020-03-19T11:38:02.6285087+11:00 [1] SUCCESS AnalyseLongRecording - Analysis Complete.
Source=S1-SP_20170925_013000.wav
Output=D:\Output\DoctorsIndices\S1-SP_20170925_013000.wav\Towsey.Acoustic
2020-03-19T11:38:02.6441289+11:00 [1] INFO  LogFileOnly - Programs stats:
{
  "Platform": "Microsoft Windows NT 6.2.9200.0",
  "ProcessorCount": 4,
  "ExecutionTime": 4.4766628,
  "PeakWorkingSet": 87810048,
  "AppConfigOverridden": false
}
2020-03-19T11:38:02.6441289+11:00 [1] INFO  LogFileOnly - ERRORLEVEL: 0
</details>

### The **operating system** I'm using is:  

Windows 10 Version 1909  

### The version of **AnalysisPrograms.exe** I'm using is:
20.2.0.99

### The data used can be found:

Data is 30 second acoustic sample every 30 minutes. There are 96 files (over a 48 hour period)

### The command I used:

```
 .\AP\AnalysisPrograms.exe ConcatenateIndexFiles Output\DoctorsIndices -o Output\DoctorsConcat2 --time-span-offset-hint "10:00" --gap-rendering NoGaps -fcs .\Configs\SpectrogramFalseColourConfig.yml -ip .\Configs\IndexPropertiesConfig.yml
```

### The config file I used:

<details><summary>A copy paste of the console output</summary>
<p>
PS D:\> .\AP\AnalysisPrograms.exe ConcatenateIndexFiles Output\DoctorsIndices -o Output\DoctorsConcat2 --time-span-offset-hint "10:00" --gap-rendering NoGaps -fcs .\Configs\SpectrogramFalseColourConfig.yml -ip .\Configs\IndexPropertiesConfig.yml
QUT Ecoacoustics Analysis Programs - version 20.2.0.99 (RELEASE build, 2020-02-11T04:22:19.0353321Z)
Git branch-version: master-8bc900890b804d5262fcecae2f05c34a9c5bb533, DirtyBuild:False, CI:000
Copyright 2020 QUT
[2020-03-19T11:49:42.0828663+11:00] WARN  -
!
!   THIS IS A BETA COMMAND.
!   It generally works but only for very narrow scenarios. Your mileage *will* vary.
!
!   DO NOT USE THE OUTPUT INDICES FOR QUANTITATIVE ANALYSIS.
!
# CONCATENATES CSV FILES CONTAINING SPECTRAL and SUMMARY INDICES.
#    IT IS ASSUMED THESE WERE OBTAINED FROM MULTIPLE SHORT AUDIO RECORDINGs
# DATE AND TIME: 19/03/2020 11:49:42 AM
# Index.csv files are in directories:
    D:\Output\DoctorsIndices
# Output directory: D:\Output\DoctorsConcat2
# DIRECTORY FILTER = *.wav

# Subdirectories Count = 96
# IndexFiles.csv Count = 96

# Start date = 24/09/2017 4:30:00 AM +10:00
# End   date = 26/09/2017 4:00:00 AM +10:00
# Time Zone  = 10:00:00
# WARNING: A sunrise/sunset data file does not exist for time zone >> 10:00:00
[2020-03-19T11:49:42.1674803+11:00] WARN  - FileStemName was empty had a default value of `DoctorsIndices` was used
# Day  count = 3

</p>
</details>

## Screenshots

![Screenshot Doctors Swamp](https://user-images.githubusercontent.com/62320779/77025802-286f5180-69e6-11ea-9f8d-bac9486f3b87.JPG)

This is the result I am getting for a 24 hour period of 30 second samples every 30 mins

**NOTE: Edited by @atruskie for brevity**

**Comment by atruskie** [comment]
Hi @joderetic,

Is this not expected behaviour?

For 30 seconds every 30 minutes, for 24 hours, we'll get 48 sets of indices for 24 hours. At ICS=30.0, that's 1 pixel wide for every set of indices, which is 48 pixels worth of data.

The `NoGaps` option literally removes any gaps in the image where there is missing data. It produces a temporally inconsistent image but has the benefit of being _more understandable_ in some cases.

There is an existing bug we know of where the x-axis is not rendered correctly. See #248.

To summarize: it looks to me like the data is being rendered correctly but the axis is not. If that's the case, I'll close this as a duplicate of #248.

**Comment by joderetic** [comment]
Thanks Anthony

Will try it

Jo

**Comment by atruskie** [comment]
Closing as duplicate of #248. Reopen if you feel this is inaccurate.
